src/librustc_mir/interpret/place.rs

```diff
@@ -1082,17 +1082,21 @@ where
                 }
```

Comment:
Oh, btw, this check seems pointless, it's impossible for `variant_index` to be out of range (without rustc being broken), so you could just ICE here if you wanted to (but there's not really any point, `dest.layout.for_variant(variant_index)` would panic trying to index with it, anyway).